src/engine/blkstore/blkstore.hpp

```diff
@@ -336,14 +336,17 @@ class BlkStore {
     BlkAllocStatus alloc_blk(const uint32_t size, blk_alloc_hints& hints, std::vector< BlkId >& out_blkid) {
         // Allocate a block from the device manager
         assert(size % m_pagesz == 0);
-        blk_count_t nblks{static_cast< blk_count_t >(size / m_pagesz)};
+        // Using a 16-bit value (blk_count_t) for page counts can lead to overflow issues when managing large storage
+        // devices. With a page size of 4096 bytes, a 16-bit value limits the addressable storage to 256 MB (65,536
+        // pages * 4096 bytes/page).
+        uint32_t nblks{static_cast< uint32_t >(size / m_pagesz)};
         if (nblks <= BlkId::max_blks_in_op()) {
             return (m_vdev.alloc_blk(nblks, hints, out_blkid));
```

Comment:
line 344 will convert uint32_t to uint16_t because m_vdev.alloc_blk takes 16bit as first parameter, which is fine here, becase we only are in if branch when nblks <= BlkId::max_blks_in_op(), but it is better to put a comment and a debug assert here by saying BlkId::max_blks_in_op() must be <= `std::numeric_limits<decltype<blk_count_t>>::max()`. 

Also please put a static_cast before nblks, when passing to alloc_blk, for the ease of read.